osquery/remote/enrollment/plugins/remote_enrollment.cpp

```diff
@@ -0,0 +1,86 @@
+#include <osquery/enrollment.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+#include <boost/property_tree/json_parser.hpp>
+
+#define MAX_TRIES 5
+
+namespace osquery {
+
+FLAG(string,
+     enrollment_uri,
+     "Not Specified",
+     "The endpoint for server side client enrollment");
+
+FLAG(string,
+     enrollment_secret,
```

Comment:
We need to also add `app_id`, for use with OAuth applications. We can leave `enrollment_secret` named as is though.